core/base-service/openapi.js

```diff
@@ -332,4 +332,34 @@ function category2openapi(category, services) {
   return spec
 }
 
-export { category2openapi }
+function pathParam({
+  name,
+  example,
+  schema = { type: 'string' },
+  description,
+}) {
+  return { name, in: 'path', required: true, schema, example, description }
+}
+
+function queryParam({
+  name,
+  example,
+  schema = { type: 'string' },
+  required = false,
+  description,
+}) {
+  if (example === null && schema.type === 'boolean') {
+    return {
+      name,
+      in: 'query',
+      required,
+      schema,
+      allowEmptyValue: true,
+      example,
+      description,
+    }
+  }
+  return { name, in: 'query', required, schema, example, description }
```

Comment:
does the order of the fields matter for consumers of this function? i know it's a huge nit, but if not, it seems like this could be deduped a bit since it's just `allowEmptyValue` being conditionally added to the returned object

Reply:
I've pushed 5a5ceb4 to tidy this up